src/vm/CMakeLists.txt

```diff
@@ -7,6 +7,9 @@ include_directories(${ARCH_SOURCES_DIR})
 add_definitions(-DUNICODE)
 add_definitions(-D_UNICODE)
 
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+add_compile_options(-Wno-address)
+endif()
```

Comment:
Can we fix the source instead? See https://dmalcolm.fedorapeople.org/gcc/2015-08-31/rst-experiment/options-to-request-or-suppress-warnings.html#cmdoption-Waddress

Reply:
This is the warning. Let me know:

In file included from src/inc/stgpool.h:28:0,
                 from src/inc/../md/inc/metamodel.h:18,
                 from src/inc/../md/inc/metamodelro.h:19,
                 from src/inc/metadata.h:17,
                 from src/vm/util.hpp:19,
                 from src/vm/common.h:110,
                 from src/vm/commodule.cpp:6:
src/vm/commodule.cpp: In function ‘Object* GetTypesInner(Module*)’:
src/vm/commodule.cpp:1066:29: warning: the address of ‘throwable’ will never be NULL [-Waddress]
         EX_CATCH_THROWABLE(&throwable);
		 
		 
src/vm/invokeutil.cpp: In static member function ‘static void InvokeUtil::SetValidField(CorElementType, TypeHandle, FieldDesc*, OBJECTREF*, OBJECTREF*, TypeHandle, CLR_BOOL*)’:
src/vm/invokeutil.cpp:978:29: warning: the address of ‘Throwable’ will never be NULL [-Waddress]
         EX_CATCH_THROWABLE(&Throwable);
                             ^
src/inc/ex.h:1087:21: note: in definition of macro ‘EX_CATCH_THROWABLE’
         if (NULL != ppThrowable)                                                \
                     ^
src/vm/invokeutil.cpp: In static member function ‘static OBJECTREF InvokeUtil::GetFieldValue(FieldDesc*, TypeHandle, OBJECTREF*, TypeHandle, CLR_BOOL*)’:
src/vm/invokeutil.cpp:1186:29: warning: the address of ‘Throwable’ will never be NULL [-Waddress]
         EX_CATCH_THROWABLE(&Throwable);
                             ^
src/inc/ex.h:1087:21: note: in definition of macro ‘EX_CATCH_THROWABLE’
         if (NULL != ppThrowable)                                                \

Reply:
Looking at `invokeutil.cpp` and `commodule.cpp`, the *address* of `Throwable` can certainly not be `NULL`; we can simply remove those three lines?

Reply:
Yes, please remove the check. We always pass it a reference to a local to this macro and I don't see why we would want to pass it NULL.

Reply:
done